src/components/v5/common/ActionSidebar/partials/forms/StagedPaymentForm/partials/StagedPaymentRecipientsField/hooks.tsx

```diff
@@ -0,0 +1,88 @@
+import { type ColumnDef, createColumnHelper } from '@tanstack/react-table';
+import React, { useMemo } from 'react';
+import { useFormContext } from 'react-hook-form';
+
+import { MAX_MILESTONE_LENGTH } from '~constants';
+import useWrapWithRef from '~hooks/useWrapWithRef.ts';
+import { formatText } from '~utils/intl.ts';
+import AmountField from '~v5/common/ActionSidebar/partials/AmountField/AmountField.tsx';
+import FormTextareaBase from '~v5/common/Fields/TextareaBase/FormTextareaBase.tsx';
+
+import PaymentBuilderPayoutsTotal from '../../../PaymentBuilderForm/partials/PaymentBuilderPayoutsTotal/index.ts';
+
+import {
+  type StagedPaymentRecipientsTableModel,
+  type StagedPaymentRecipientsFieldModel,
+} from './types.ts';
+
+export const useStagedPaymentTableColumns = (
+  name: string,
+  data: StagedPaymentRecipientsFieldModel[],
+) => {
+  const { watch } = useFormContext();
```

Comment:
I think it will be better to make separate PR/issue for this because there is a lot of `useFormContext` uses 🥲 
<img width="161" alt="image" src="https://github.com/JoinColony/colonyCDapp/assets/76940796/7eb7666e-f321-49b4-84c9-b1381fb44507">

Reply:
I'm sorry, I think I worded that wrong -- I was thinking we can strictly type the areas within your PR, just  your PR, that uses this `useFormContext` hook. I made similar changes in one of my other PRs basically, I think it's nice because it makes it less error prone 😄